crates/re_viewer/src/app_state.rs

```diff
@@ -0,0 +1,224 @@
+use ahash::HashMap;
+
+use re_data_store::StoreDb;
+use re_log_types::{ApplicationId, LogMsg, StoreId, TimeRangeF};
+use re_smart_channel::Receiver;
+use re_viewer_context::{
+    AppOptions, Caches, ComponentUiRegistry, PlayState, RecordingConfig, SpaceViewClassRegistry,
+    ViewerContext,
+};
+use re_viewport::ViewportState;
+
+use crate::ui::Blueprint;
+
+const WATERMARK: bool = false; // Nice for recording media material
+
+#[derive(Default, serde::Deserialize, serde::Serialize)]
+#[serde(default)]
+pub struct AppState {
+    /// Global options for the whole viewer.
+    app_options: AppOptions,
+
+    /// Things that need caching.
+    #[serde(skip)]
+    pub(crate) cache: Caches,
+
+    #[serde(skip)]
+    selected_rec_id: Option<StoreId>,
+    #[serde(skip)]
+    pub(crate) selected_blueprint_by_app: HashMap<ApplicationId, StoreId>,
+
+    /// Configuration for the current recording (found in [`StoreDb`]).
+    recording_configs: HashMap<StoreId, RecordingConfig>,
+
+    selection_panel: crate::selection_panel::SelectionPanel,
+    time_panel: re_time_panel::TimePanel,
+
+    // TODO(jleibs): This is sort of a weird place to put this but makes more
+    // sense than the blueprint
+    #[serde(skip)]
+    viewport_state: ViewportState,
+}
+
+impl AppState {
+    /// The selected/visible recording id, if any.
+    pub fn recording_id(&self) -> Option<StoreId> {
+        self.selected_rec_id.clone()
+    }
+
+    /// The selected/visible recording id, if any.
+    pub fn set_recording_id(&mut self, recording_id: StoreId) {
+        self.selected_rec_id = Some(recording_id);
+    }
+
+    pub fn app_options(&self) -> &AppOptions {
+        &self.app_options
+    }
+
+    pub fn app_options_mut(&mut self) -> &mut AppOptions {
+        &mut self.app_options
+    }
+
+    /// Currently selected section of time, if any.
+    #[cfg_attr(target_arch = "wasm32", allow(dead_code))]
+    pub fn loop_selection(&self) -> Option<(re_data_store::Timeline, TimeRangeF)> {
+        self.selected_rec_id.as_ref().and_then(|rec_id| {
+            self.recording_configs
+                .get(rec_id)
+                // is there an active loop selection?
+                .and_then(|rec_cfg| {
+                    rec_cfg
+                        .time_ctrl
+                        .loop_selection()
+                        .map(|q| (*rec_cfg.time_ctrl.timeline(), q))
+                })
+        })
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn show(
```

Comment:
It was already confusing to me that `show` is implemented on `AppState` rather than being implemented on `App`, but moving it out of `app.rs` into `app_state.rs` makes that organization even more obviously awkward.

Reply:
Yeah, I agree. I hope to simplify things a bit more in a future PR